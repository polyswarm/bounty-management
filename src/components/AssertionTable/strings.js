const strings = {
  author: 'Author',
  verdict: 'Verdict',
  metadata: 'Metadata',
  bid: 'Bid',
};

export default strings;
