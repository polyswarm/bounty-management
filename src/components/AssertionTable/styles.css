.AssertionTable {
  color: black;
  overflow: auto;
  width: 100%;
}

table {
  border-collapse: collapse;
}

thead {
  background-color: #8500ff;
  color: white;
  height: 60px;
}

tr {
  border-bottom: 1px solid #a5a5a5;
}

td {
  height: 40px;
  text-align: left;
}

th {
  box-sizing: border-box;
  height: 60px;
  padding-bottom: 5px;
  text-align: left;
  vertical-align: bottom;
}
