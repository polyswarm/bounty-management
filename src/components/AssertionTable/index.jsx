import React, { Component } from 'react';
import PropTypes from 'prop-types';
// Bounty imports
import AssertionRow from '../AssertionRow';
// Component imports
import strings from './strings';
import './styles.css';

class AssertionTable extends Component {

  render() {
    const { props: { assertions } } = this;
    return (
      <table className='AssertionTable' cellPadding='10'>
        <thead>
          <tr>
            <th>{strings.author}</th>
            <th>{strings.verdict}</th>
            <th>{strings.metadata}</th>
            <th>{strings.bid}</th>
          </tr>
        </thead>
        <tbody>
          {
            assertions.map((assertion) => {
              return (
                <AssertionRow
                  key={assertion.author+assertion.bid+assertion.metadata+assertion.verdict}
                  assertion={assertion}/>
              );
            })
          }
        </tbody>
      </table>
    );
  }

}
AssertionRow.proptypes = {
  assertions: PropTypes.array.isRequired,
};
export default AssertionTable;
