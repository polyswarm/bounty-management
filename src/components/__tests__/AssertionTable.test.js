import React from 'react';
import {render} from 'enzyme';
import {renderToJson} from 'enzyme-to-json';
import AssertionTable from '../AssertionTable';

it('renders without crashing', () => {
  const assertions = [];
  const wrapper = render(<AssertionTable assertions={assertions}/>);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});

it('renders the correct number of children', () => {
  const assertions = [
    {author: 'asdf', verdict: false, bid: 5, metadata: 'Nothing'},
    {author: 'demo', verdict: false, bid: 5, metadata: ''},
    {author: 'fdsa', verdict: true, bid: 1000, metadata: 'VIRUS'},
  ];
  const wrapper = render(<AssertionTable assertions={assertions}/>);

  expect(wrapper.find('.AssertionRow')).toHaveLength(3);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});
