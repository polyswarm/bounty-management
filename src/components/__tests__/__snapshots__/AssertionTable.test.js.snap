// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the correct number of children 1`] = `
<table
  class="AssertionTable"
>
  <tbody>
    <tr>
      <th>
        Author
      </th>
      <th>
        Verdict
      </th>
      <th>
        Metadata
      </th>
      <th>
        Bid
      </th>
    </tr>
    <tr
      class="AssertionRow"
    >
      <td>
        asdf
      </td>
      <td
        class="Assertion-Benign"
      >
        Benign
      </td>
      <td>
        Nothing
      </td>
      <td>
        5
      </td>
    </tr>
    <tr
      class="AssertionRow"
    >
      <td>
        demo
      </td>
      <td
        class="Assertion-Benign"
      >
        Benign
      </td>
      <td>
        
      </td>
      <td>
        5
      </td>
    </tr>
    <tr
      class="AssertionRow"
    >
      <td>
        fdsa
      </td>
      <td
        class="Assertion-Malignant"
      >
        Malignant
      </td>
      <td>
        VIRUS
      </td>
      <td>
        1000
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`renders without crashing 1`] = `
<table
  class="AssertionTable"
>
  <tbody>
    <tr>
      <th>
        Author
      </th>
      <th>
        Verdict
      </th>
      <th>
        Metadata
      </th>
      <th>
        Bid
      </th>
    </tr>
  </tbody>
</table>
`;
